test/dates/io.jl

```diff
@@ -267,8 +291,7 @@ f = "y m d"
 @test_throws ArgumentError Dates.Date("1 1 32",f)
 @test_throws ArgumentError Dates.Date(" 1 1 32",f)
 @test_throws ArgumentError Dates.Date("# 1 1 32",f)
-# can't find 1st space delimiter,s o fails
-@test_throws ArgumentError Dates.Date("1",f)
+@test Dates.Date("1",f) == Dates.Date(1)
```

Comment:
Thanks for the answers here - sounds reasonable. The behavior changes make this not a backport candidate though.